src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/MigrateXprojProjectFactory.cs

```diff
@@ -0,0 +1,142 @@
+
+            // TODO: we need to read the output from the migration report in addition to console output.
+            // We'll still want to read console output in case of a bug in the dotnet cli, and it errors with some exception
+            // https://github.com/dotnet/roslyn-project-system/issues/507
+            var output = process.StandardOutput.ReadToEnd();
+            var err = process.StandardError.ReadToEnd();
+
+            if (!string.IsNullOrEmpty(output))
+            {
+                pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_INFORMATIONAL, projectName, bstrFileName, output);
+            }
+
+            if (!string.IsNullOrEmpty(err))
+            {
+                pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_WARNING, projectName, bstrFileName, err);
+            }
+
+            if (process.ExitCode == 0)
+            {
+                success = true;
+
+                // TODO: We need to find the newly created csproj. This will only be necessary until dotnet migrate adds a Migration Report.
+                // https://github.com/dotnet/roslyn-project-system/issues/507
+                var files = Directory.EnumerateFiles(directory);
+                pbstrUpgradedFullyQualifiedFileName = files.FirstOrDefault(file => file.EndsWith(".csproj"));
+                if (string.IsNullOrEmpty(pbstrUpgradedFullyQualifiedFileName))
+                {
+                    pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_ERROR, projectName, bstrFileName, string.Format(VSResources.NoMigratedCSProjFound, directory));
+                    pbstrUpgradedFullyQualifiedFileName = bstrFileName;
+                    success = false;
+                }
+            }
+            else
+            {
+                pbstrUpgradedFullyQualifiedFileName = bstrFileName;
+                pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_ERROR, projectName, bstrFileName,
+                    string.Format(VSResources.XprojMigrationFailed, projectName, directory, bstrFileName, process.ExitCode));
+                success = false;
+            }
+
+            return success ? VSConstants.S_OK : VSConstants.VS_E_PROJECTMIGRATIONFAILED;
+        }
+
+        public int UpgradeProject_CheckOnly(string bstrFileName, IVsUpgradeLogger pLogger, out int pUpgradeRequired, out Guid pguidNewProjectFactory, out uint pUpgradeProjectCapabilityFlags)
+        {
+            var isXproj = bstrFileName.EndsWith(".xproj");
+
+            // If the project is an xproj, then we need to one-way upgrade it. If it isn't, then there's nothing we can do with it.
+            pUpgradeRequired = isXproj ? (int)__VSPPROJECTUPGRADEVIAFACTORYREPAIRFLAGS.VSPUVF_PROJECT_ONEWAYUPGRADE :
+                                         (int)__VSPPROJECTUPGRADEVIAFACTORYREPAIRFLAGS.VSPUVF_PROJECT_NOREPAIR;
+
+            if (isXproj)
+            {
+                pguidNewProjectFactory = new Guid($"{{{CSharpProjectSystemPackage.ProjectTypeGuid}}}");
```

Comment:
Double curlies escape the curly in interpolated strings. This is equivalent to `"{" + CSharpProjectSystemPackage.ProjectTypeGuid + "}"`.